src/spikeinterface/core/base.py

```diff
@@ -128,8 +128,18 @@ def ids_to_indices(
                 indices = np.arange(len(self._main_ids))
         else:
             assert isinstance(ids, (list, np.ndarray, tuple)), "'ids' must be a list, np.ndarray or tuple"
+
+            non_existent_ids = [id for id in ids if id not in self._main_ids]
```

Comment:
I always find it a bit dangerous to shadow a python function built-in even in a list comprehension. But should be okay I guess.

Also might be safer to cast this to a bool no? sometimes we end up with weird list situations so relying on an empty list seems a bit more dangerous. But up to you.

Reply:
Ah, good suggestion about the shadowing. I actually totally forgot that id is a built-in. 

Concerning the empty list it is pretty much part of the spec:
https://docs.python.org/3/library/stdtypes.html#truth-value-testing

Why do you think it would be dangerous?

Reply:
Oh, let's keep the ids because it is used just right here under:

https://github.com/h-mayorquin/spikeinterface/blob/e2b1a3b734245e696b8e83d73354a62161f8fcff/src/spikeinterface/core/base.py#L139-L142

And I rather be consistent as comprehension has it own scope so I am not that concerned about the other side-effects.

I would rather merge this (if people accept it) and then do a second PR to get rid of the shadowing of id all at once.

Reply:
I think for the empty list it was fine, but I remember some weird boolean thing with an empty array or an array with a 0. So just forcing to a bool would prevent that. I just didn't want us to forget there are some weird edges cases for some containers. But I think you're right it was an empty list it was an array issue.

Reply:
> Oh, let's keep the ids because it is used just right here under:

Works for me. I just like to keep it in mind. List comprehensions have such a small scope they are fine, I'm always more worried when we use built-ins at the function level.

Reply:
This:

```python
is_one_equal_to_0 = np.all((1 == 0 for _ in range(10)))
if is_one_equal_to_0:
    print("This is great, 1 is equal to 0")
```